internals/secrethub/env_source.go

```diff
@@ -116,6 +131,12 @@ func (env *environment) env() (map[string]value, error) {
 	referenceEnv := newReferenceEnv(osEnvMap)
 	sources = append(sources, referenceEnv)
 
+	// --secrets-dir flag
```

Comment:
When it comes to precedence of the different options to pass environment variables, I'd rather make the `--secrets-dir` flag of lower precedence. Because it passes a bunch of environment variables at once, I think you'd want the other options to override those.

I'm suggesting the following order:
1. `--envar`
2. secret reference syntax (secrethub://path/to/secret)
3. `.env` files. By default `secrethub.env` is used; Use the `--env-file` flag to use a different file.
4. `--secrets-dir`
5. envvars passed by the OS

Reply:
Good point! I updated the precedence rules.